From 3ad6cba1ae540622afbe1f8250af23b2e1a51cde Mon Sep 17 00:00:00 2001
From: Alexandru Gagniuc <mr.nuke.me@gmail.com>
Date: Sun, 18 May 2025 21:11:16 -0500
Subject: [PATCH] arm64: dts: qcom: ipq9574: add NSSNOC interconnects to nsssc
 node

Add the interconnects to the nsscc clock controller node. I don't
remember exactly what problem I was solving. Without this change, the
ethernet PPE driver gets an -EINVAL from phyling when connecting to a
QCA8084 phy. Exact mechanism unknown, but this fixes it.

Signed-off-by: Alexandru Gagniuc <mr.nuke.me@gmail.com>
---
 arch/arm64/boot/dts/qcom/ipq9574.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

--- a/arch/arm64/boot/dts/qcom/ipq9574.dtsi
+++ b/arch/arm64/boot/dts/qcom/ipq9574.dtsi
@@ -1300,6 +1300,11 @@
 			#clock-cells = <1>;
 			#reset-cells = <1>;
 			#interconnect-cells = <1>;
+
+
+			interconnects = <&gcc MASTER_NSSNOC_NSSCC &gcc SLAVE_NSSNOC_NSSCC>,
+					<&gcc MASTER_NSSNOC_SNOC_0 &gcc SLAVE_NSSNOC_SNOC_0>,
+					<&gcc MASTER_NSSNOC_SNOC_1 &gcc SLAVE_NSSNOC_SNOC_1>;
 		};
 
 		qcom_ppe: ethernet@3a000000 {
